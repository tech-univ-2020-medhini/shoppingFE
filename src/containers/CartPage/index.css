.cart-page{
    display: flex;
    justify-content: center;
    flex-direction: column;
    flex:1;
}
.cart-text{
    display: flex;
    align-self: center;
    padding-top: 2rem;
    padding-bottom: 2rem;
}
.cart-page-display{
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    align-content: center;
}
.card-and-button{
    display: flex;
    flex-direction: column;
}