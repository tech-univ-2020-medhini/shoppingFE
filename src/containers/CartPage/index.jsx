import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { useHistory } from 'react-router-dom';
import Table from '../../components/Table';
import TotalCard from '../../components/TotalCard';
import Button from '../../components/Button';
import './index.css';

const CartPage = (props) => {
  const history = useHistory();
  const { cartCount } = props;
  const [categories, setCategories] = useState([]);
  const [productList, setProductList] = useState([]);
  const filteredProducts = {};
  let totalCost = 0;
  useEffect(() => {
    const getProductList = async () => {
      const productResponse = await axios.get('http://localhost:8080/products');
      setProductList(productResponse.data);
      const categoriesResponse = await axios.get('http://localhost:8080/categories');
      setCategories(categoriesResponse.data);
      productList.forEach((product) => {
        if (product.cart !== 0) {
          filteredProducts[product.category].push(product);
          totalCost += product.cart * product.price;
        }
      });
    };
    getProductList();
  }, []);
  const onCheckout = async () => {
    const patchQuantities = async () => {
      const response = await axios.patch('http://localhost:8080/quantities', productList);
      history.push('/');
    };
    patchQuantities();
  };
  const onContinueShopping = () => {
    history.push('/');
  };
  return (
    <div className="cart-page">
      <div className="text">
        Your Shopping Cart (
        {cartCount}
        {' '}
        items)
      </div>
      <Table filteredProducts={filteredProducts} />
      <TotalCard price={totalCost} onClickHandler={onCheckout} />
      <Button buttonColor="grey" text="Continue Shopping" onClickHandler={onContinueShopping} />
    </div>
  );
};
CartPage.propTypes = {
  cartCount: PropTypes.number.isRequired,
};

export default CartPage;
