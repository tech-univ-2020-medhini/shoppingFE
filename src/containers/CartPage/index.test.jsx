import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import CartPage from './index';


describe('the cart page container', () => {
  const history = createMemoryHistory();
  it('should render correctly', () => {
    const { asFragment } = render(<Router history={history}><CartPage /></Router>);

    expect(asFragment()).toMatchSnapshot();
  });
});
