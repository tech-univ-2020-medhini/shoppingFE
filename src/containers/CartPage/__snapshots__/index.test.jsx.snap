// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the cart page container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="cart-page"
  >
    <div
      class="cart-text"
    >
      Your Shopping Cart (0 items)
    </div>
    <div
      class="cart-page-display"
    >
      <div
        class="table"
      >
        <table
          border="1"
        >
          <tbody>
            <tr>
              <th>
                Item
              </th>
              <th>
                Price
              </th>
              <th>
                Quantity
              </th>
              <th>
                Subtotal
              </th>
            </tr>
          </tbody>
          <tbody />
        </table>
      </div>
      <div
        class="card-and-button"
      >
        <div
          class="total-card"
        >
          <div
            class="total-card-text"
          >
            <div
              class="total"
            >
              Total
            </div>
            <div
              class="rs"
            >
              RS
            </div>
            <div
              class="price"
            >
              0
            </div>
          </div>
          <hr />
          <div
            class="total-checkout"
          >
            <button
              class="button-yellow"
              data-testid="test-btn"
              type="button"
            >
              Checkout
            </button>
          </div>
        </div>
        <button
          class="button-white"
          data-testid="test-btn"
          type="button"
        >
          Continue Shopping
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
